Objective-C/CBLDatabase.mm

```diff
@@ -969,4 +1015,50 @@ - (BOOL) saveResolvedDocument: (CBLDocument*)resolvedDoc
     return YES;
 }
 
+
+# pragma mark DOCUMENT EXPIRATION
+
+
+- (void) scheduleDocumentExpiration: (NSTimeInterval)minimumDelay {
+    [self cancelPreviousPerformRequestsForPurgingExpiredDocuments];
+    
+    UInt64 nextExpiration = c4db_nextDocExpiration(_c4db);
+    if (nextExpiration > 0) {
+        NSDate* expDate = [NSDate dateWithTimeIntervalSince1970: nextExpiration];
+        NSTimeInterval delay = MAX(expDate.timeIntervalSinceNow + 1.0, minimumDelay);
```

Comment:
I'm not certain if adding an extra one second is needed.

Reply:
I will check this.

Reply:
removed the extra buffer time, and all tests are still passing.